tests/testthat/test-drive_list.R

```diff
@@ -8,97 +8,133 @@ context("List files")
+               verbose = FALSE)
+  rm <- file.remove("chickwts.txt")
+}
+
+
 test_that("drive_list() not confused by same-named folders", {
   skip_on_appveyor()
   skip_on_travis()
 
-  foo_id <- drive_mkdir("foo")$id
-  bar_id <- drive_mkdir("bar", path = "foo")$id
-  baz_id <- drive_mkdir("baz", path = "foo/bar")$id
-  yo_id <- drive_mkdir("yo", path = "foo/bar/baz")$id
-  bar_2_id <- drive_mkdir("bar", path = "foo")$id
-
   ## there should be two folders named 'bar' in 'foo'
-  expect_true(all(c(bar_id, bar_2_id) %in% drive_list(path = "foo")$id))
+  expect_true(all(c(nm_("bar"), nm_("bar")) %in% drive_list(path = nm_("foo"))$name))
 
   ## there should be no trouble telling which bar to route through
-  expect_identical(yo_id, drive_list(path = "foo/bar/baz")$id)
-
-  ## clean up
-  foo_id %>%
-    drive_file() %>%
-    drive_delete()
+  expect_identical(nm_("yo"),
+                   drive_list(path = file.path(nm_("foo"),
+                                               nm_("bar"),
+                                               nm_("baz"))
+                   )
+                   $name)
 
 })
 
 test_that("get_leaf() errors when two distinct folders have same path", {
   skip_on_appveyor()
   skip_on_travis()
 
-  ## create foo/bar
-  foo_id <- drive_mkdir("foo")$id
-  bar_id <- drive_mkdir("bar", path = "foo")$id
-
-  ## create another foo/bar
-  foo_2_id <- drive_mkdir("foo")$id
-
-  ## our drive_mkdir won't let you place bar in foo, since it doesn't know which foo to place
-  ## it in, so we will use plain httr to make the second foo/bar
-  bar_2 <- httr::POST("https://www.googleapis.com/drive/v3/files",
-                      drive_token(),
-                      body = list(
-                        name = "bar",
-                        parents = list(foo_2_id),
-                        mimeType = "application/vnd.google-apps.folder"
-                      ),
-                      encode = "json"
-  )
-  bar_2_id <- process_response(bar_2)$id
-
   expect_error(
-    get_leaf("foo/bar"),
-    "The path 'foo/bar' identifies more than one file:"
+    get_leaf(file.path(nm_("foo"), nm_("bar"))),
+    paste0("The path '",
+           file.path(nm_("foo"), nm_("bar")),
+           "' identifies more than one file:")
   )
-
-  ## clean up
-  clean <- c(foo_id, foo_2_id) %>%
-    purrr::map(drive_file) %>%
-    purrr::map(drive_delete)
 })
 
 test_that("get_leaf() is not confused by same-named leafs at different depths", {
 
   skip_on_appveyor()
   skip_on_travis()
 
-  # +-- foo
-  # | +-- bar
-  # +-- bar
-  foo_id <- drive_mkdir("foo")$id
-  bar_id <- drive_mkdir("bar", path = "foo/")$id
-  bar_2_id <- drive_mkdir("bar")$id
-  expect_identical(get_leaf("foo/bar/")$id, bar_id)
-
-  c(foo_id, bar_2_id) %>%
-    purrr::map(drive_file) %>%
-    purrr::map(drive_delete)
+  expect_length(get_leaf(file.path(nm_("foo"),
+                                   nm_("bar"),
+                                   nm_("baz"))),
+                3)
```

Comment:
I think I changed it to `baz` because I already used `bar` in that place for a different test so since I'm not deleting & recreating them each time, I had to pick a different name - but I'll double check. 👍🏼 For `expect_silence`